### PR TITLE
[ci skip] Enhanced rdoc for Hash

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: hash.c [hunk 1]
```diff
@@ -2150,15 +2152,16 @@ rb_hash_stlike_lookup(VALUE hash, st_data_t key, st_data_t *pval)
 
 /*
  *  call-seq:
- *     hsh[key]    ->  value
+ *    hsh[key] -> value
```
Comment (ioquatix) [inline hash.c:2155]: I don't know how common it is, but do you think we should avoid abbreviations?

Comment (BurdetteLamar) [reply hash.c:2155]: That's what's been here in Hash, and 'ary' is seen everywhere over at Array.  I'm going to revise all of Hash, so we could change 'em all as we go.

Btw, Matz has approved me as a committer, so as soon as I figure out the security stuff (SSH, etc.), I'll be merging my own stuff.

Comment (ioquatix) [reply hash.c]: You should still get reviews from other members of the group.

Comment (BurdetteLamar) [reply hash.c]: Count on it.  I'd never do otherwise.

**Pull request merged**